docker/hoodie/hadoop/sparkworker/pom.xml

```diff
@@ -30,6 +30,7 @@
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <checkstyle.skip>true</checkstyle.skip>
+    <main.basedir>${project.parent.parent.basedir}</main.basedir>
```

Comment:
please test once if you are able to indepdently run a mvn command from one of the sub modules. i.e `cd hudi-cli; mvn clean install` should work..

Reply:
Good catch.
Before this PR(such as in master branch), we could run `mvn clean install -pl hudi-client` in project root dir to install specified module but not run `mvn clean install` in sub modules dirs i.e hudi-client(Unable to find configuration file
 at location F:/community/incubator-hudi/hudi-client/style/scalastyle-config.xml). I just updated this PR to make it works and verified `mvn clean install -DskipTests` in sub modules(hudi-cli, hudi-client, hudi-spark..).

Reply:
And maybe we would seperate the commit to another PR(this PR should based on) since it is not so relative to this PR. WDYT?

Reply:
ideally yes. but as long as building independent modules work, I am okay for this PR. your call. Let me know how you want to proceed

Reply:
I was thinking to open another PR (may also need to define main.basedir as this PR does) to make mvn clean install works in sub modules(just a minor git history improvement). However, it would also be done in this PR. So please merge this PR if there is no other concern.